app/samples/phase_detection/src/flight.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <array>
+#include <cstdint>
+#include <f_core/flight/CPhaseController.h>
+
+/**
+ * Flight Events for a fictional flight
+ */
+enum Events : uint8_t { PadReady, Boost, Coast, Noseover, MainChute, GroundHit, CamerasOff, NumEvents };
+constexpr std::array<const char *, Events::NumEvents> eventNames = {
+    "PadReady", "Boost", "Coast", "Noseover", "MainChute", "GroundHit", "CamerasOff",
+};
+
+/**
+ * Sources of flight events
+ */
+enum Sources : uint8_t {
+    IMU1,
+    Barom1,
+    NoseoverLockout,
+    Boost2CoastTimer,
+    Noseover2MainTimer,
+    FullFlightTimer,
+    ExtraCameraTimer,
+    NumSources
+};
+inline constexpr std::array<const char *, Sources::NumSources> sourceNames = {
+    "IMU 1 (LSM6DSL)",     "Barom 1 (BME280)",  "Noseover Lockout",  "Boost2Coast Timer",
+    "Noseover2Main Timer", "Full Flight Timer", "ExtraCamera Timer",
+};
+
+inline constexpr std::size_t num_timer_events = 5;
+using Controller = PhaseController<Events, Events::NumEvents, Sources, Sources::NumSources, num_timer_events>;
+
+/**
+ * Special events triggered not by sensors but by timers between phases
+ */
+inline constexpr std::array<Controller::TimerEvent, num_timer_events> timer_events = {
+    // The engine should burn for around X seconds. don't detect coast unless the engine has been firing for a bit
+    // Can be implemented as a lockout or as another way of progressing states if you don't want to do unboost detection
+    Controller::TimerEvent{
```

Comment:
Syntax for these is 
start: When this event happens, start counting
event: When the timer expires report this event
time: How long to wait
source: What source is reporting the event